### DISCUSSION TITLE
What is useReducer hook in React?

### DISCUSSION THREAD

**Discussion by PranthRaval** [HarshitKumar9030/Tailwind-Components#9] [Q&A] [answered]
### **What is useReducer hook in React?**
Please tell me about it I found it really hard to understand

**Comment by HarshitKumar9030** [comment, marked as answer]
Hii Pranth, useReducer is basically an alternative to useState, It accepts a reducer of type (state, action) => newState, and returns the current state paired with a dispatch method. (If you’re familiar with Redux, you already know how this works.)

You can refer to these url's to know more,

-   https://www.w3schools.com/react/react_usereducer.asp
- https://reactjs.org/docs/hooks-reference.html#usereducer

_**Thanks**_
